addon/object/index.js

```diff
@@ -147,11 +147,11 @@ export const action = decorator(function(target, key, desc) {
  * @param {...String} propertyNames - List of property keys this computed is dependent on
  */
 export const computed = decoratorWithParams(function(target, key, desc, params) {
-  if (!desc.writable) {
-    throw new Error('ember-decorators does not support using getters and setters');
+  if (desc.value instanceof Ember.ComputedProperty) {
```

Comment:
So it turns out this is actually the behavior defined in the decorator spec: http://tc39.github.io/proposal-decorators/#sec-updated-semantics

Step 22: Let coalesced be the new List created by combining getters and setters with the same [[Key]]. If both a getter and setter with the same [[Key]] have a non-empty [[Decorators]], throw a new Error.

So if both the getter _and_ setter are decorated, that's a syntax error, which makes sense.